code/game/atoms.dm

```diff
@@ -535,6 +535,12 @@ its easier to just keep the beam vertical.
 /atom/proc/shuttle_act(var/datum/shuttle/S)
 	return
 
+/atom/proc/clean_act(var/cleanliness)//1 = contact with water (splashed with water, removes glue from objs), 2 = space cleaner or efficient cleaning (showers, sink, soap), 3 = bleach
+	if (cleanliness >= CLEANLINESS_SPACECLEANER)
+		clean_blood()
+	if (cleanliness >= CLEANLINESS_BLEACH)
+		color = ""
```

Comment:
But what about buses or anything else that could change something's color?

Reply:
Same as before, bleach would reset it. Though now it's properly handled instead of being snowflaked in bleach reaction code.